docs/en/integrations/upload-integration.asciidoc

```diff
@@ -0,0 +1,50 @@
+[[upload-a-new-integration]]
+= Upload an integration to Kibana
+
+++++
+<titleabbrev>Upload an integration</titleabbrev>
+++++
+
+{fleet} supports integration installation through direct upload as a means to support integration developers
+or users who have created custom integrations that they don't want to commit upstream back to the https://github.com/elastic/integrations[Elastic Integrations repository].
+
+Direct upload can also be useful in air-gapped environments,
+by providing a way to update integrations without needing to update a self-hosted package registry.
+
+[discrete]
+[[upload-integration-local]]
+== Local development
+
+If you've followed the local development steps in <<build-a-new-integration>>, upload your integration to Kibana with the following command:
+
+[source,terminal]
+----
+elastic-package install --zip /path/to/my/custom-integration
+----
+
+For more information, see <<elastic-package-install>>.
+
+[discrete]
+[[upload-integration-production]]
+== Production deployment
+
+To upload your integration to a production deployment, first zip the package:
+
+[source,terminal]
+----
+$ zip -r my-custom-integration.zip /path/to/my/custom-integration
```

Comment:
It'd be better to use `elastic-package build` to create the zip. It does some transformations on the packaged files that are required in some packages.
```suggestion
$ cd /path/to/my/custom-integration
$ elastic-package build
```